### PR TITLE
Added jsdoc comments.

### PR DESCRIPTION
**Pull request by mistakenot**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -5,7 +5,14 @@ const setImmediate = require('async/setImmediate')
...
 module.exports = Block
 
-// Immutable block of data
+/**
+ * Represents an immutable block of data that is uniquely referenced with a multihash key.
+ *
+ * @constructor
+ * @param {Buffer | string} data - The data to be stored in the block as a buffer or a UTF8 string.
+ * @example
+ * var block = new Block('a012d83b20f9371...');
```
Comment (dignifiedquire) [inline src/index.js]: Please use the same style for examples we use in the rest of the code, (no semicolons, prefer `const` and `let` over `var`)

File: src/index.js [hunk 2]
```diff
@@ -36,6 +43,18 @@ function Block (data) {
...
+  * Creates a unique multihash key of this block.
+  *
+  * @param {string} [hashFunc='sha2-256'] - The hash function to use.
+  * @param {function} callback - The callback to execute on completion.
+  * @returns {void}
+  * @example
+  * block.key((hashkey) => {
+  *   console.log(hashkey);
+  * });
+  * // 'QmeoBGh5g5kHgK3xppJ1...'
```
Comment (dignifiedquire) [inline src/index.js:56]: same as above

File: src/index.js [hunk 3]
```diff
@@ -36,6 +43,18 @@ function Block (data) {
     }
   })
 
+  /**
+  * Creates a unique multihash key of this block.
+  *
+  * @param {string} [hashFunc='sha2-256'] - The hash function to use.
+  * @param {function} callback - The callback to execute on completion.
```
Comment (dignifiedquire) [inline src/index.js]: You can indicate the function signature like this `@param {function(Error, Multihash)} callback`

**Review by dignifiedquire — CHANGES_REQUESTED** [review]
Thank you 🎉  
I left some comments for small improvements.

**Comment by mistakenot** [comment]
Thanks for the feedback, have implemented suggested changes.

As an aside, I wonder if you can get the linter to also check jsdoc example sections?

**Comment by dignifiedquire** [comment]
> As an aside, I wonder if you can get the linter to also check jsdoc example sections?

I would love to, but so far I haven't found anything.

**Pull request merged**